zebra-network/src/peer_set/set/tests/vectors.rs

```diff
@@ -0,0 +1,161 @@
+        std::mem::discriminant(&ReceiveRequestAttempt::Empty)
+    );
+
+    // Build a peerset
+    runtime.block_on(async move {
+        let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
+            .with_discover(discovered_peers)
+            .with_minimum_peer_version(minimum_peer_version.clone())
+            .build();
+
+        // Get a ready future
+        let peer_ready_future = peer_set.ready();
+
+        // Drop the future
+        std::mem::drop(peer_ready_future);
+
+        // Peer set will remain ready for requests
+        let peer_ready1 = peer_set
+            .ready()
+            .await
+            .expect("peer set service is always ready");
+
+        // Make sure the client did not received anything yet
+        assert_eq!(
+            std::mem::discriminant(&client_handle.try_to_receive_outbound_client_request()),
+            std::mem::discriminant(&ReceiveRequestAttempt::Empty)
+        );
+
+        // Make a call to the peer set that returns a future
+        let fut = peer_ready1.call(Request::Peers);
+
+        // Client received the request
+        match client_handle.try_to_receive_outbound_client_request() {
+            ReceiveRequestAttempt::Request(client_request) => {
+                assert_eq!(client_request.request, Request::Peers)
+            }
+            _ => unreachable!(),
+        };
+
+        // Drop the future
+        std::mem::drop(fut);
+
+        // Peer set will remain ready for requests
+        let peer_ready2 = peer_set
+            .ready()
+            .await
+            .expect("peer set service is always ready");
+
+        // Get a new future calling a different request than before
+        let _fut = peer_ready2.call(Request::MempoolTransactionIds);
+
+        // Client received the request
+        match client_handle.try_to_receive_outbound_client_request() {
+            ReceiveRequestAttempt::Request(client_request) => {
+                assert_eq!(client_request.request, Request::MempoolTransactionIds)
+            }
+            _ => unreachable!(),
+        };
+    });
+}
+
+#[test]
+fn peer_set_ready_multiple_connections() {
+    // Use three peers with the same version
+    let peer_version = Version::min_specified_for_upgrade(Network::Mainnet, NetworkUpgrade::Canopy);
+    let peer_versions = PeerVersions {
+        peer_versions: vec![peer_version, peer_version, peer_version],
+    };
+
+    // Start the runtime
+    let runtime = zebra_test::init_async();
+    let _guard = runtime.enter();
+
+    // Get peers and client handles of them
+    let (discovered_peers, handles) = peer_versions.mock_peer_discovery();
+    let (minimum_peer_version, _best_tip_height) =
+        MinimumPeerVersion::with_mock_chain_tip(Network::Mainnet);
+
+    // Make sure we have the right number of peers
+    assert_eq!(handles.len(), 3);
+
+    // Build a peerset
+    runtime.block_on(async move {
+        let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
```

Comment:
Nit: Moving the comment to be closer to the code block.

```suggestion
    runtime.block_on(async move {
        // Build a peerset
        let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/3252/commits/55cb55e2549e842db84b62373e2afa983ce2a76e